models/publication.go

```diff
@@ -1,10 +1,85 @@
 package models
 
-import "time"
+import (
+	"errors"
+	"strconv"
+	"time"
+)
 
 type PublicationId int64
 
 type Publication struct {
 	AuthorId     UserId    `json:"authorId"`
 	CreationTime time.Time `json:"creationTime"`
 }
+
+var NoNotesToPublishError = errors.New("There are no unpublished notes to publish")
+
+func (db *DB) PublishNotes(userId UserId) error {
+
+	myUnpublishedNotes, err := db.GetMyUnpublishedNotes(userId)
+	if err != nil {
+		return err
+	}
+
+	if len(myUnpublishedNotes) == 0 {
+		return NoNotesToPublishError
+	}
+
+	publicationId, err := db.StoreNewPublication(&Publication{AuthorId: userId, CreationTime: time.Now().UTC()})
+
+	sqlQuery := `
+		INSERT INTO note_to_publication_relationship (publication_id, note_id) 
+		VALUES ($1, $2)`
+
+	noteIds := make([]int64, len(myUnpublishedNotes))
+
+	{
+		i := 0
+		for noteId := range myUnpublishedNotes {
+			noteIds[i] = int64(noteId)
+			i++
+		}
+	}
+
+	values := make([]interface{}, len(noteIds)*2, len(noteIds)*2)
```

Comment:
This is equivalent to `make([]interface{}, len(noteIds)*2)`. The shorter form is more common/canonical. The long form is mostly useful if you want to set the capacity of a slice to be greater than the length.